### DISCUSSION TITLE
MDX syntax for Markdown text?

### DISCUSSION THREAD

**Discussion by ZenaMel** [facebook/docusaurus#8753] [Q&A]
Use case:

Given something like this:

```md
| a column             | another column                                                                                                  |
| -------------------- | --------------------------------------------------------------------------------------------------------------- |
| a little bit of text | reaaaaaaaaaaaaaaaaaaaaaally long text with multiple lines of content and admonitions, **markdown syntax**, etc. |
```

I'd like to add things like an admonition or other things into the table.

To make it easier to manage I was thinking of doing something like:

```md
| a column             | another column     |
| -------------------- | ------------------ |
| a little bit of text | <AnotherColumn1 /> |

export const AnotherColumn1 = () => (`
reaaaaaaaaaaaaaaaaaaaaaally long text with multiple lines of content and admonitions, **markdown syntax**, etc.
`)
```

But I'm having trouble parsing the content as Markdown.
Wrapping it with MDXContent like so doesn't work (it's just handled as regular text like the rest):

```md
| a column             | another column     |
| -------------------- | ------------------ |
| a little bit of text | <AnotherColumn1 /> |

export const AnotherColumn1 = () => (`
<MDXContent>
reaaaaaaaaaaaaaaaaaaaaaally long text with multiple lines of content and admonitions, **markdown syntax**, etc.
<MDXContent />
`)
```

Is there a workaround to this problem or even an idiomatic way in MDX?

**Comment by slorber** [comment]
This works on the MDX 2 playground (make sure to turn on the remark-gfm option)

https://mdxjs.com/playground/

<img width="1023" alt="CleanShot 2023-03-09 at 17 01 00@2x" src="https://user-images.githubusercontent.com/749374/224081386-bde8c24f-1017-495f-ad60-26396a27df79.png">


I'm not sure you can use markdown syntax inside an inline JSX component. A React component does not use markdown syntax by default so a component declared in MDX probably does not either. You could import the MDX content from another file instead.

**Reply by slorber** [reply]
Maybe @wooorm can confirm if it's possible or not, but I'm not too surprised the 2nd case doesn't work (exported React component or element):

<img width="987" alt="CleanShot 2023-03-09 at 17 07 44@2x" src="https://user-images.githubusercontent.com/749374/224083402-7caf9877-6712-41ff-88d3-aaca5ea3134f.png">

**Reply by wooorm** [reply]
You can’t use markdown inside JS. It’s the other way around. JS exists in markdown

**Reply by slorber** [reply]
Thanks that's the intuition I had :)

**Reply by ZenaMel** [reply]
Thanks for the insight!

Would you have a different work around in mind for what I'm trying to do? Perhaps I could write a custom component that wraps MDXContent and call it inside MDX?

**Reply by slorber** [reply]
> Would you have a different work around in mind 

The only thing we have currently to render markdown is MDX, at compile time. So you really need to go through the Webpack loader and create a standalone MDX file if you want to make it work, or you'd have to provide your own runtime Markdown renderer.

Later, with React Server Components, we could enable you parse/render markdown at runtime in a more dynamic way without having the cost of shipping a md processor to the browser. This could be useful for this use-case, but also for rendering MD anywhere including in landing pages.

Docusaurus v1 had this: https://v1.docusaurus.io/docs/en/api-pages#complibrarymarkdownblock

<img width="817" alt="CleanShot 2023-03-10 at 11 56 32@2x" src="https://user-images.githubusercontent.com/749374/224298550-6c2dee5c-462f-49e9-ae5d-0d571cd54435.png">

We didn't include this in v2 because this time React is hydrated client-side and we don't want to ship a Markdown parser/renderer by default when most sites don't need it. But you can still provide it yourself if you really want to.

For example, use https://github.com/remarkjs/react-markdown, but keep in mind that it's not a tiny lib. Server Components would mitigate this problem but for now if you want that you have to pay the extra cost of including this lib in your browser bundle.

**Reply by wooorm** [reply]
I believe JSX solves the issue:

```jsx
export function AnotherColumn1() {
  return <>
    reaaaaaaaaaaaaaaaaaaaaaally long text with multiple lines of content{' '}
    and admonitions, <strong>markdown syntax</strong>, etc.
  </>
}
```

You can also use JSX inside MDX for your custom table needs:

```mdx
<MyTable>
  <MyTableRow>
    <MyTableCell>reaaaaaaaaaaaaaaaaaaaaaally long text with multiple
    lines of content and admonitions, **markdown syntax**, etc.</MyTableCell>
  </MyTableRow>
</MyTable>
```

**Reply by ZenaMel** [reply]
> I believe JSX solves the issue:
> 
> ```js
> export function AnotherColumn1() {
>   return <>
>     reaaaaaaaaaaaaaaaaaaaaaally long text with multiple lines of content{' '}
>     and admonitions, <strong>markdown syntax</strong>, etc.
>   </>
> }
> ```
> 
> You can also use JSX inside MDX for your custom table needs:
> 
> ```gfm
> <MyTable>
>   <MyTableRow>
>     <MyTableCell>reaaaaaaaaaaaaaaaaaaaaaally long text with multiple
>     lines of content and admonitions, **markdown syntax**, etc.</MyTableCell>
>   </MyTableRow>
> </MyTable>
> ```

This would be in HTML/JSX, not Markdown syntax though. I was hoping I could define things like bullet lists etc. without having to rely on the more verbose HTML syntax.

**Reply by wooorm** [reply]
My second example shows how to do that
